graphql-database-manager-test/src/test/java/com/fleetpin/graphql/database/manager/test/DynamoDbRevisionTest.java

```diff
@@ -0,0 +1,251 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.fleetpin.graphql.database.manager.test;
+
+import java.util.Arrays;
+import java.util.concurrent.ExecutionException;
+
+import org.junit.jupiter.api.Assertions;
+
+import com.fleetpin.graphql.database.manager.Database;
+import com.fleetpin.graphql.database.manager.RevisionMismatchException;
+import com.fleetpin.graphql.database.manager.Table;
+import com.fleetpin.graphql.database.manager.test.annotations.DatabaseNames;
+import com.fleetpin.graphql.database.manager.test.annotations.TestDatabase;
+
+public class DynamoDbRevisionTest {
+	
+	@TestDatabase
+	public void testCreateNewObject(final Database db) throws InterruptedException, ExecutionException {
+		var entry = new SimpleTable("1", "garry");
+		db.checkPut(entry).get();
+		var entries = db.query(SimpleTable.class).get();
+		Assertions.assertEquals(1, entries.size());
+		Assertions.assertEquals("garry", entries.get(0).name);
+		
+		var e = new SimpleTable("1", "garry");
+		var cause = Assertions.assertThrows(ExecutionException.class, () -> db.checkPut(e).get());
+		
+		Assertions.assertEquals(RevisionMismatchException.class, cause.getCause().getClass());
+	}
+	
+	@TestDatabase
+	public void testRevisionMustStartAt0(final Database db) throws InterruptedException, ExecutionException {
+		var entry = new SimpleTable("1", "garry");
+		entry.setRevision(1);
+		//does not exist yet so fails
+		var cause = Assertions.assertThrows(ExecutionException.class, () -> db.checkPut(entry).get());
+		Assertions.assertEquals(RevisionMismatchException.class, cause.getCause().getClass());
+	}
+
+	@TestDatabase
+	public void testIncrementsAfterExists(final Database db) throws InterruptedException, ExecutionException {
+		var entry = new SimpleTable("1", "garry");
+		
+		entry = db.checkPut(entry).get();
+		entry = db.checkPut(entry).get();
```

Comment:
Maybe assert the revision after each put? It at least helps me follow what sort of state the row is in.

Reply:
done